src/include/migraphx/matcher.hpp

```diff
@@ -646,6 +664,19 @@ MIGRAPHX_PRED_MATCHER(broadcast, instruction_ref ins)
     return contains({"broadcast", "multibroadcast"}, ins->name());
 }
 
+/*
+ * Input and output shapes have the same rank.  This is assumed for broadcast
+ * instructions for some fusions.
+ */
+MIGRAPHX_PRED_MATCHER(dims_match, instruction_ref ins)
```

Comment:
Let's name this `ndims_match` so we're consistent. `dims_match` makes it sound like you're matching that each dimension of the shape is the same (checking the `shape.lens()`).

Reply:
Probably better to name as `input_output_ndim_match`